### PR TITLE
Igor lig 302 write benchmark code for imagenet

### PR DESCRIPTION
**Pull request by IgorSusmelj**
## Changes

- add benchmark for ImageNet

### CODE REVIEW

**Comment by Atharva-Phatak** [comment]
@IgorSusmelj I am wondering are you getting the same results as the published paper ? or with are +- 2% range ?.

**Comment by IgorSusmelj** [comment]
@Atharva-Phatak, I'm also looking forward to seeing the results :)
Will keep you posted here!

**Comment by IgorSusmelj** [comment]
I spent quite some time tuning parameters for some of the models such as SimCLR on ImageNet. It looks like the following are rather crucial (more crucial than I thought):

- learning rate
- learning rate warmup
- weight decay
- temperature

It's rather important to use the numbers from the paper. I somehow thought +- with standard parameters you'd get close enough accuracy. But training for 50 epochs would just get to ~17% test accuracy.

**Comment by IgorSusmelj** [comment]
Currently training a ResNet-18 model using SimCLR for 100 epochs to see whether the parameters match. So far we are at 24% test accuracy. I'm afraid something might still be off. Will push the current state of the code soon.
```
Epoch 28:  34%|███▎      | 871/2600 [13:49<27:26,  1.05it/s, loss=1.36, v_num=14, lr=0.553, kNN_accuracy=23.90]
```

**Comment by IgorSusmelj** [comment]
Started a new run using a ResNet-50. 
- `10` epochs linear learning rate warmup as in the paper
- lr is `0.3`
- batch size is `256`
- temperature is `0.1`
- precision is `fp16`
- weight decay is `1e-6`

```
Epoch 8: 100%|██████████| 5200/5200 [1:08:18<00:00,  1.27it/s, loss=1.25, v_num=15, lr=0.240, kNN_accuracy=18.30]
```

**Comment by IgorSusmelj** [comment]
Looks like with 100-NN and MoCov2 we should get the orange plot. Can use that as a rough reference on what to expect. (from this paper: https://arxiv.org/pdf/2111.12309.pdf)
image

**Comment by IgorSusmelj** [comment]
Closed in favor of https://github.com/lightly-ai/lightly/pull/1162
